filtered_conn.go

```diff
@@ -23,11 +44,14 @@ func (c *filteredConn) ReadFrom(b []byte) (n int, addr net.Addr, rerr error) {
 		if n < 1 || b[0]&0x80 == 0 {
 			return
 		}
-		maddr, err := toQuicMultiaddr(addr)
+		rmAddr, err := toQuicMultiaddr(addr)
 		if err != nil {
 			panic(err)
 		}
-		if !c.filters.AddrBlocked(maddr) {
+
+		connAddrs := &connAddrs{lmAddr: c.lmAddr, rmAddr: rmAddr}
+
+		if c.gater.InterceptAccept(connAddrs) {
 			return
 		}
```

Comment:
This is interesting. I don't think it translates to the new model properly. Our `InterceptAccept` is supposed to be called once, when we decide to accept or reject an incoming connection. The previous implementation is calling the filters _on every single network packet_. Have a look through #124 -- there's some history there.

@marten-seemann -- do you recall why incoming filtering wasn't implemented [here instead](https://github.com/libp2p/go-libp2p-quic-transport/blob/master/reuse.go#L183)? We could call the `InterceptAccept` and `Close()` the UDP connection if rejected. Alternatively if we want to avoid the kernel notifying us new packets from the same source as new UDP connections, we could just "drain" the connection, which is effectively what we're doing now anyway.

Reply:
`newReuseConn` is called once per UDP "connection". That is, once per address we're listening on.

In this case, we call `InterceptAccept` once per packet that starts with `0x80`. Those are packets with the QUIC "long" header. Usually, these packets are only sent for the initial handshake.

The alternative would be to add low-level hooks into the QUIC transport itself before QUIC completes the crypto handshake.

Reply:
Right, I had a senior moment and got confused by golang's use of the term "Connection" to refer to a UDP listener.